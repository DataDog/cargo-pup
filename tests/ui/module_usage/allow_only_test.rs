//@compile-flags: --crate-name allow_only
//@compile-flags: --crate-type lib

// Test demonstrating the AllowOnly rule type of module_usage lint

// Module with imports that should only allow certain modules
mod test_allow_only {
    // These imports are allowed
    use std::fmt;
    use std::io;

    // This import is not in the allowed list and should trigger the lint
    use std::collections::HashMap; //~ ERROR: Use of module 'std::collections::HashMap' is not allowed; only ["std::fmt", "std::io"] are permitted.

    // This import is also not allowed
    use std::env; //~ ERROR: Use of module 'std::env' is not allowed; only ["std::fmt", "std::io"] are permitted.

    fn test_function() {
        let mut map = HashMap::new();
        map.insert("key", "value");
        let _ = std::env::var("HOME");
    }
}
