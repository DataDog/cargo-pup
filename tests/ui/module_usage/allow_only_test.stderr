error: Use of module 'std::collections::HashMap' is not allowed; only ["std::fmt", "std::io"] are permitted.
  --> tests/ui/module_usage/allow_only_test.rs:13:5
   |
LL |     use std::collections::HashMap;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Consider removing or changing the import.
   = note: Applied by cargo-pup rule 'allow_only_rules'.
   = note: `#[deny(module_usage)]` on by default

error: Use of module 'std::env' is not allowed; only ["std::fmt", "std::io"] are permitted.
  --> tests/ui/module_usage/allow_only_test.rs:16:5
   |
LL |     use std::env;
   |     ^^^^^^^^^^^^^
   |
   = help: Consider removing or changing the import.
   = note: Applied by cargo-pup rule 'allow_only_rules'.

error: aborting due to 2 previous errors

